lib/kube/proxy/single_cert_handler_test.go

```diff
@@ -0,0 +1,166 @@
+	encKubeCluster := base64.RawURLEncoding.EncodeToString([]byte(kubeCluster))
+	restConfig.Host += fmt.Sprintf("/v1/teleport/%s/%s", encTeleportCluster, encKubeCluster)
+
+	client, err := kubernetes.NewForConfig(restConfig)
+	require.NoError(t, err)
+
+	return client
+}
+
+func TestSingleCertRouting(t *testing.T) {
+	kubeMockA, err := testingkubemock.NewKubeAPIMock()
+	require.NoError(t, err)
+	t.Cleanup(func() { kubeMockA.Close() })
+
+	kubeMockB, err := testingkubemock.NewKubeAPIMock(
+		// This endpoint returns a known mock error so we can determine from the
+		// response which cluster the request was routed to.
+		testingkubemock.WithGetPodError(
+			metav1.Status{
+				Status:  metav1.StatusFailure,
+				Message: "cluster b error",
+				Reason:  metav1.StatusReasonInternalError,
+				Code:    http.StatusInternalServerError,
+			},
+		),
+	)
+	require.NoError(t, err)
+	t.Cleanup(func() { kubeMockB.Close() })
+
+	defaultRoleSpec := RoleSpec{
+		Name:       roleName,
+		KubeUsers:  roleKubeUsers,
+		KubeGroups: roleKubeGroups,
+	}
+
+	const clusterName = "root.example.com"
+
+	tests := []struct {
+		name string
+
+		kubeClusterOverride string
+		roleSpec            RoleSpec
+		assert              func(t *testing.T, restConfig *rest.Config)
+	}{
+		{
+			name:     "successful path routing to multiple clusters",
+			roleSpec: defaultRoleSpec,
+			assert: func(t *testing.T, restConfig *rest.Config) {
+				clientA := pathRoutedKubeClient(t, restConfig, clusterName, "a")
+				_, err := clientA.CoreV1().Pods(metav1.NamespaceDefault).List(context.Background(), metav1.ListOptions{})
+				require.NoError(t, err)
+
+				clientB := pathRoutedKubeClient(t, restConfig, clusterName, "b")
+				_, err = clientB.CoreV1().Pods(metav1.NamespaceDefault).List(context.Background(), metav1.ListOptions{})
+				require.NoError(t, err)
+			},
+		},
+		{
+			name:     "cannot access nonexistent cluster",
+			roleSpec: defaultRoleSpec,
+			assert: func(t *testing.T, restConfig *rest.Config) {
+				client := pathRoutedKubeClient(t, restConfig, clusterName, "c")
+				_, err = client.CoreV1().Pods(metav1.NamespaceDefault).List(context.Background(), metav1.ListOptions{})
+				require.ErrorContains(t, err, "not found")
+			},
+		},
+		{
+			name: "cannot access cluster denied by roles",
+			roleSpec: RoleSpec{
+				Name:       roleName,
+				KubeUsers:  roleKubeUsers,
+				KubeGroups: roleKubeGroups,
+				SetupRoleFunc: func(r types.Role) {
+					r.SetKubeResources(types.Deny, []types.KubernetesResource{{Kind: types.KindKubePod, Name: types.Wildcard, Namespace: types.Wildcard, Verbs: []string{types.Wildcard}}})
+				},
+			},
+			assert: func(t *testing.T, restConfig *rest.Config) {
+				client := pathRoutedKubeClient(t, restConfig, clusterName, "a")
+				_, err = client.CoreV1().Pods(metav1.NamespaceDefault).List(context.Background(), metav1.ListOptions{})
+				require.ErrorContains(t, err, "cannot list resource")
```

Comment:
I've noticed a mild behavior change here, I think since this just wasn't possible before.  Attempting to e.g. list resources on a cluster denied by roles now yields:
```
Error from server (Forbidden): kubernetes cluster "default" not found
```

...however, attempting to list a nonexistent cluster returns:
```
Error from server (NotFound): the server could not find the requested resource
```

(and a `Kubernetes cluster "$name" not found"` is logged on the server)

Would we consider this a resource name oracle?